query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -429,6 +447,46 @@ public Observable<Row> handleRequest(
     }
   }
 
+  private void measurePinotTagQueryExecutionTime(
+      ExecutionContext executionContext, Stopwatch stopwatch) {
+    try {
+      Set<String> referencedColumns = executionContext.getReferencedColumns();
+      Preconditions.checkNotNull(referencedColumns);
+      if (referencedColumns.contains(TAG_COLUMN)) {
+        pinotTagQueryExecutionTimer.record(stopwatch.elapsed());
+      }
+    } catch (Exception e) {
+      LOG.error("Error occurred while measuring PinotTagQueryExecutionTime: ", e);
+    }
+  }
+
+  private void measureRequestAge(ExecutionContext executionContext) {
+    try {
+      Preconditions.checkNotNull(executionContext);
+      Preconditions.checkNotNull(executionContext.getQueryTimeRange());
+      if (executionContext.getQueryTimeRange().isPresent()) {
+        Optional<QueryTimeRange> queryAge = executionContext.getQueryTimeRange();
+        if (queryAge.isPresent()) {
```

Comment:
I think this condition will always be true. So, you can remove this `if` and put an `assert queryRange.isPresent()`.

Reply:
isn't assert only meant for tests? In codebase also assert is being used only in tests.

Reply:
Done